lib/awesome_nested_set/model.rb

```diff
@@ -79,9 +79,13 @@ def right_of(node)
           end
 
           def nested_set_scope(options = {})
-            options = {:order => { order_column_name => :asc }}.merge(options)
+            order = scope_order_from_options(options)
+            default_scoped.where(options[:conditions]).order(order)
+          end
 
-            where(options[:conditions]).order(options.delete(:order))
+          def nested_set_scope_without_default_scope(options = {})
+            order = scope_order_from_options(options)
+            unscoped.where(options[:conditions]).order(order)
           end
 
```

Comment:
I know this is an old release by now, but I believe this change breaks the behavior of `root` and `roots` when there's already an established scope on the ActiveRecord relation. `roots` does not pass any of the scope options along to `nested_set_scope`, so that information is lost when you use `default_scoped` and `unscoped` instead of `where`.

For example, if I have an association between `User` and `Comment` where comments are a nested set, `user.comments.roots` would previously return all root comments that belong to the user. Now it returns *all root comments by any user*. This isn't intended behavior, is it?

@marcrohloff @aualdrich sorry if I'm not going through the right channels, but I ran into this issue while upgrading this gem. If it's more appropriate to make an Issue I can do that, but perhaps this change is 100% intentional and I'm just misusing Awesome Nested Set, so I wanted to better understand why this change was made.

Reply:
Sorry, it looks like this was addressed in https://github.com/collectiveidea/awesome_nested_set/pull/437 . I first ran into this issue a couple months ago, and it seems that since then the issue has been resolved. 😅